learning_resources_search/serializers.py

```diff
@@ -372,9 +373,33 @@ class SearchResponseMetadata(TypedDict):
 
 class SearchResponseSerializer(serializers.Serializer):
     count = serializers.SerializerMethodField()
+    next = serializers.SerializerMethodField()
+    previous = serializers.SerializerMethodField()
     results = serializers.SerializerMethodField()
     metadata = serializers.SerializerMethodField()
 
+    def construct_pagination_url(self, instance, request, link_type="next"):
+        if request:
+            url = request.build_absolute_uri()
+            total_record_count = self.get_count(instance)
+            offset = int(request.query_params.get("offset", 0))
+            limit = int(request.query_params.get("limit", 20))
+            if link_type == "previous":
+                offset -= limit
+            else:
+                offset += limit
+            if offset >= 0 and offset <= total_record_count:
+                return replace_query_param(url, "offset", offset)
+        return None
+
+    def get_next(self, instance) -> int:
+        request = self.context.get("request")
+        return self.construct_pagination_url(instance, request, link_type="next")
+
+    def get_previous(self, instance) -> int:
```

Comment:
```suggestion
    def get_previous(self, instance) -> str:
```
and similarly in `get_next`. 

Note: In addition to type hints in editors, this affects the schema that [drf-spectacular](https://github.com/tfranzel/drf-spectacular) generates